### PR TITLE
WT-9578 Implement the bounded cursor API for the Dump cursor

### PR DESCRIPTION
**Pull request by jiechenbo**

### CODE REVIEW

**Review by luke-pearson — APPROVED** [review]
LGTM.

File: src/cursor/cur_dump.c [hunk 1]
```diff
@@ -323,6 +323,19 @@ __curdump_set_value(WT_CURSOR *cursor, ...)
         return (child->op(child));                 \
     }
 
+/*
+ * __curdump_bound --
+ *     WT_CURSOR::search_near for dump cursors.
```
Comment (luke-pearson) [inline src/cursor/cur_dump.c:328]: Fix this comment.

File: test/suite/test_cursor_bound16.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        dumpcurs = self.session.open_cursor(self.uri + self.file_name,
+                                                None, dumpopt)
+
+        # Set bounds at lower key 30 and upper key at 50.
+        self.set_bounds(dumpcurs, self.gen_dump_key(30), "lower")
+        self.set_bounds(dumpcurs, self.gen_dump_key(50), "upper")
+        self.cursor_traversal_bound(dumpcurs, self.gen_dump_key(30), self.gen_dump_key(50), True, 21)
+        self.cursor_traversal_bound(dumpcurs, self.gen_dump_key(30), self.gen_dump_key(50), False, 21)
+        
+        # Test basic search near scenarios.
```
Comment (clarissecheah) [inline test/suite/test_cursor_bound16.py:87]: Fix this comment too.

**Review by clarissecheah — APPROVED** [review]
LGTM with np.

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**